lib/internal/errors.js

```diff
@@ -1095,6 +1095,12 @@ E('ERR_INVALID_PACKAGE_TARGET', (pkgPath, key, subpath, target) => {
   if (key === '.') {
     return `Invalid "exports" main target ${JSONStringify(target)} defined ` +
       `in the package config ${pkgPath}${sep}package.json`;
+  } else if (typeof target === 'string' && target !== '' &&
+      (!target.startsWith('./') || !target.startsWith('/'))) {
+    return `Invalid "exports" target ${JSONStringify(target)} defined for '${
+      StringPrototypeSlice(key, 0, -subpath.length || key.length)}' in the ` +
+      `package config ${pkgPath}${sep}package.json. Targets must start \`./\`` +
+      'or `/`';
```

Comment:
I don't think we allow mappings that start with a slash because they would go all the way back to the file system root by URL resolution semantics.

```suggestion
  } else if (typeof target === 'string' && target !== '' &&
      !target.startsWith('./')) {
    return `Invalid "exports" target ${JSONStringify(target)} defined for '${
      StringPrototypeSlice(key, 0, -subpath.length || key.length)}' in the ` +
      `package config ${pkgPath}${sep}package.json. ` +
      'Targets must start with `./`';
```